app/models/user_subscription.rb

```diff
@@ -4,12 +4,30 @@
 class UserSubscription < ApplicationRecord
   ALLOWED_TYPES = %w[Article].freeze
 
-  belongs_to :author, class_name: "User", inverse_of: :user_subscriptions
-  belongs_to :subscriber, class_name: "User", inverse_of: :user_subscriptions
+  belongs_to :author, class_name: "User", inverse_of: :source_authored_user_subscriptions
+  belongs_to :subscriber, class_name: "User", inverse_of: :subscribed_to_user_subscriptions
   belongs_to :user_subscription_sourceable, polymorphic: true
 
   validates :author_id, presence: true
-  validates :subscriber_id, presence: true, uniqueness: { scope: %i[user_subscription_sourceable_type user_subscription_sourceable_id] }
+  validates :subscriber_email, presence: true
+  validates :subscriber_id, presence: true, uniqueness: { scope: %i[subscriber_email user_subscription_sourceable_type user_subscription_sourceable_id] }
   validates :user_subscription_sourceable_id, presence: true
   validates :user_subscription_sourceable_type, presence: true, inclusion: { in: ALLOWED_TYPES }
+
+  def self.make_new(source:, subscriber:)
```

Comment:
Could it make sense to follow e.g. `FactoryBot` naming and use `build` here? `make_new` is less intent revealing, especially when next to `make`.